lib/rihanna/job.ex

```diff
@@ -177,6 +177,65 @@ defmodule Rihanna.Job do
   end
 
   @doc false
+  def delete_by(opts) do
+    ids_to_delete =
+      opts
+      |> filter_term_list()
+      |> Enum.reduce("", fn [id, _term], acc ->
+        "#{acc}'#{id}',"
+      end)
+      |> String.replace_trailing(",", "")
+
+    with false <- ids_to_delete == "",
```

Comment:
This `with` is very confusing. What's wrong with using a simple `if`?

Reply:
Personal preference to avoid nested `if`s and `case`s, but I'll see if I can make it better